### PR TITLE
feature: provide a way for plugins to get the list of serial ports

### PR DESCRIPTION
**Pull request by sbender9**
This includes making so `plugin.schema()` and `plugin.schemaUI()` can return a promise

### CODE REVIEW

**Comment by sbender9** [comment]
Example usage here: https://github.com/sbender9/signalk-serial-ports-example-plugin

**Review by tkurki — CHANGES_REQUESTED** [review]
I started down the rabbit hole by looking into why there are no ts bindings for @serialport/binddings, making that @ts-ignore needed. 

And I realised that I've probably introduced a wrong way to get a serialport listing.

Docs at https://serialport.io/docs/api-stream#serialportlist say that you should
```
const SerialPort = require('serialport')
SerialPort.list().then(x => console.log(x))
```
so it looks like instead of adding to the injury we should replace both instances of @serialport/bindings with the proper listing method.

File: src/interfaces/plugins.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ import Debug from 'debug'
 import { Request, Response } from 'express'
 const debug = Debug('signalk:interfaces:plugins')
 // @ts-ignore
+import SerialPort from 'serialport'
+// @ts-ignore
```
Comment (tkurki) [inline src/interfaces/plugins.ts]: I think there are types for serialport, installing them should get rid of the need for the ignore and then we can return something other than any below

File: src/serverroutes.js [hunk 2]
```diff
@@ -567,50 +567,12 @@ module.exports = function(app, saveSecurityConfig, getSecurityConfig) {
...
-    Promise.all([
-      listSafeSerialPortsDevSerialById(),
-      listSafeSerialPortsDevSerialByPath(),
-      listSafeSerialPortsOpenPlotter(),
-      listSerialPorts()
-    ])
-      .then(([byId, byPath, byOpenPlotter, serialports]) =>
-        res.json({ byId, byPath, byOpenPlotter, serialports })
-      )
+    listAllSerialPorts().then(ports => {
```
Comment (tkurki) [inline src/serverroutes.js]: no need for the curly braces

**Comment by sbender9** [comment]
I don't understand why lint is failing. I did not change anything with that line of code.  What am I'm missing??

**Comment by tkurki** [comment]
You moved that from .js to .ts - differerent rules apply.

**Comment by tkurki** [comment]
There's fs.promises.readdir since 10.0.0, time to use that?

**Comment by sbender9** [comment]
Shoot I was looking at the wrong file... (plugins.ts), let me switch to fs.promises...

**Pull request merged**